compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -796,6 +797,140 @@ static FailureOr<Flow::WorkloadBuilder> getWorkloadBuilder(OpBuilder &builder,
   return result;
 }
 
+/// Prune loops with 1 trip count
+static int64_t pruneUnitTripLoops(llvm::ArrayRef<int64_t> loopRanges) {
+  return llvm::count_if(loopRanges, [](int64_t range) { return range > 0; });
+}
+
+/// Finds contiguous loop ranges with iterator type "parallel" from the given
+/// linalg.generic.
+/// TODO: Current collapsing mechanism collapses the shape of the tensor not the
+/// loops in the linalg.generic, so we cannot collapse transpose like things.
+/// This function eliminates non-contiguous loop indices from the affine_map.
+static SmallVector<ReassociationIndices> getParallelContiguousLoops(
```

Comment:
I saw `CollapseDimsPass` that actually collapses reductions loops. Therefore, I did not do `reduction` loops.

I can remove `CollapseDimsPass` and collapse everything here if it makes sense?

Reply:
One reason the pass collapsing reduction is happening earlier is to be able to compose it with reduction splitting. We could merge those two but we loose the ability to split reductions or we would have to also move that during dispatch region creation.
We don't rely on this at the time but I would like us to pick a solution where this will be possible in the future.

Reply:
For a more actionable feedback: I would leave it as is in this PR.

Reply:
Sounds good, let's discuss this, this is probably not super urgent so whenever you get some down time.